modules/launcher/src/main/java/org/mule/module/launcher/log4j2/MuleLoggerContext.java

```diff
@@ -73,13 +80,24 @@
             applicationClassloader = false;
             artifactName = null;
         }
+
+        Collection<Field> candidateFields = getAllFields(LoggerContext.class, withName("loggers"));
+        if (candidateFields.size() == 1)
+        {
+            loggersField = candidateFields.iterator().next();
+            loggersField.setAccessible(true);
+        }
+        else
+        {
+            loggersField = null;
+        }
     }
 
     @Override
     public synchronized void reconfigure()
     {
-        super.reconfigure();
         applyContainerConfiguration();
+        super.reconfigure();
```

Comment:
As of log4j 2.5, the configuration has to be set before the lifcecycle management of the log4j objects (that lifecycle is hadled by the reconfigure of the superclass). This seems to have been a benign issue before the upgrade.